pkg/catalogv2/content/content.go

```diff
@@ -456,11 +457,13 @@ func isRancherAndBundledCatalog(repo repoDef) bool {
 // skipIndexFiltering returns true if the helm index filtering should be skipped.
 func skipIndexFiltering() bool {
 	serverVersion := settings.ServerVersion.Get()
+	version := rancherSemver.Version(serverVersion)
 
 	// If the server version is a release or a release candidate (rc), we don't skip filtering.
-	if settings.IsRelease() || strings.Contains(serverVersion, "-rc") {
+	if settings.ServerVersionHasReleasePrefixExcludesHead() || version.IsRC() {
```

Comment:
Both this function and `settings.ServerVersionHasReleasePrefixExcludesHead()` perform the same steps:
```
	serverVersion := ServerVersion.Get()
	version := semver.Version(serverVersion)
```

I think we should consider refactoring to remove this duplication.

```suggestion
	if (version.HasReleasePrefix() && !strings.Contains(serverVersion, "head")) || version.IsRC() {
```